### PR TITLE
Use pre-built cross-compiled aarch64 gcc for x86-64 linux machine

### PR DESCRIPTION
**Pull request by potsrevennil**
[//]: # (SPDX-License-Identifier: CC-BY-4.0)

Made x86-64 linux machine to use the pre-built corss-compile aarch64 gcc
Added x86-64 linux GitHub runner, it will only cross compile the project, emulation on qemu will be done in other PR

- **add arm gnu gcc nix spec for x86_64**
- **use the pre-built cross-compiled gcc for x86_64 machines in ci**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -8,14 +8,21 @@ on:
     branches: ["main"]
 jobs:
   build_test:
-    runs-on: macos-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+        - os: ubuntu-latest
```
Comment (hanno-becker) [inline .github/workflows/build.yml:15]: We're implicitly relying on github runners for `ubuntu-latest` to be x86_64 here, right?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -8,37 +8,72 @@ on:
...
-          ./checksum.sh ./test/gen_KAT512  ec4ac397e595ac7457cb7d8830921faf3290898a10d7dd3864aab89ea61fe9a3
-          ./checksum.sh ./test/gen_KAT768  9a0826ad3c5232dfd3b21bc4801408655c565a491b760f509b2ee2cd7180babe
-          ./checksum.sh ./test/gen_KAT1024 6dafb867599b750a6a831b03e494cf41dea748c78a0e275e7b268bbb893cf37d
+          cat >> $GITHUB_STEP_SUMMARY <<-EOF
+            ## KAT tests
+          EOF
+
+          make CROSS_PREFIX=${{ matrix.cross_prefix }} kat
+
+          if [ -z ${{ matrix.cross_prefix }} ]; then
```
Comment (hanno-becker) [inline .github/workflows/build.yml:57]: Perhaps add a `skipped` for now otherwise.

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -8,37 +8,72 @@ on:
...
-        run: |
-          make clean
+          cat >> $GITHUB_STEP_SUMMARY <<-EOF
+            ## NISTKAT tests
+          EOF
+
+          make CROSS_PREFIX=${{ matrix.cross_prefix }} nistkat
+
+          if [ -z ${{ matrix.cross_prefix }} ]; then
+            cat >> $GITHUB_STEP_SUMMARY <<-EOF
```
Comment (hanno-becker) [inline .github/workflows/build.yml:74]: Same here: Add `skipped` or similar until we run the tests in emulation.

**Comment by cothan** [comment]
I would say it's easier to install zig and perform cross-compile. ` -target x86_64-native` or ` -target aarch64-linux` and zig can be installed easily.

File: .github/workflows/build.yml (hunk 1 above)
Comment (potsrevennil) [reply .github/workflows/build.yml:15]: maybe we can check with `uname -m`, but what to do if it's not x86_64, just fail immediately ?

**Comment by potsrevennil** [comment]
zig seems like a nice solution for this!

File: .github/workflows/build.yml (hunk 1 above)
Comment (hanno-becker) [reply .github/workflows/build.yml:15]: Yes

**Comment by hanno-becker** [comment]
@cothan 

> I would say it's easier to install zig and perform cross-compile.  -target x86_64-native or  -target aarch64-linux and zig can be installed easily.

What specifically would you suggest to be changed compared to the present state of the PR?

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -11,14 +11,20 @@ jobs:
     strategy:
       matrix:
         system: [macos-latest, pqcp-arm64]
+        skipped: [false]
+        include:
+        - system: ubuntu-latest
```
Comment (hanno-becker) [inline .github/workflows/ci.yml:16]: Just to be sure I understand the meaning of this: It means we add `ubuntu-latest` to the `system` array, and set `skipped` to `false` for `macos-latest` and `pqcp-arm64`, but to `true` to `ubuntu-latest`?

**Comment by potsrevennil** [comment]
If we're using `zig` for cross compiling, other than updating the dependencies, adding something like this in the `Makefile` is all we need I think. (Then we can get rid of the `arm-gcc-gnu.nix` file that I added in this PR)

```
CC := zig cc
TARGET ?=

ifdef TARGET
        ZFLAGS += -target $(TARGET)
endif

...:
  $(CC) $(ZFLAGS) ...
```

Then in CI we will need to specify the target when cross compilation is needed (i.e. `make TARGET=aarch64-linux-gnu`)

**Review by hanno-becker — APPROVED** [review]

**Comment by hanno-becker** [comment]
@potsrevennil Is this ready to go in?

**Comment by potsrevennil** [comment]
@hanno-becker yes, if we ever want `zig` I can file another pr

File: .github/workflows/ci.yml (hunk 4 above)
Comment (potsrevennil) [reply .github/workflows/ci.yml:16]: yes, exactly

**Comment by potsrevennil** [comment]
@hanno-becker yes, I think I can file another pr for `zig` if we ever want that

**Comment by hanno-becker** [comment]
@potsrevennil Ready to merge once the DCO is fixed.

**Comment by potsrevennil** [comment]
@hanno-becker yes, this is ready for merge, I've just updated to fail the job if the architecture is not as expected.

For `zig`, I think I can file another pr if we ever want to use `zig` for cross compiling.

**Comment by hanno-becker** [comment]
Thank you @potsrevennil, merged.

Are you doing the follow-up adding qemu-based tests?

**Comment by potsrevennil** [comment]
@hanno-becker Yes, I will do the follow-up qemu-based tests, but got stucked by other issue unfortunately, so maybe will not finish it very soon

**Pull request merged**